department-of-veterans-affairs/caseflow
# Piedram/appeals 13370

Resolves [APPEALS-13367](https://vajira.max.gov/browse/APPEALS-13367), [APPEALS-13365](https://vajira.max.gov/browse/APPEALS-13365), [APPEALS-13370](https://vajira.max.gov/browse/APPEALS-13370)

### Description

As a developer I need to create the action's functionality for the "Post-Send Initial Notification Letter Holding Period" Task so that Contested Claim letter notifications can be tracked within Caseflow

### Acceptance Criteria:


##### 1- The following occurs when the “Proceed to Final Notification Letter” task action is selected for the "Post-Send Initial Notification Letter Holding Period" task (will be used so that user can go straight to the “Send Final Notification Letter” – reserved for situations when only the final notification letter needs to be resent)

- [ ] Modal appears asking user to confirm that they want to proceed to final notification letter.
- [ ] User is required to provide a reason why they want to proceed to final notification letter (Field type: free text with a 350-character limit)
- [ ] If confirm, The “Post-Send Initial Notification Letter Holding Period” task is marked as complete and Task moves from Currently Active Tasks section to Case Timeline and contains the following information
> 1 Date of occurrence
> 2 Assigned to: <organization>
> 3 Assigned by: <user who completed action to assign task to COB>
> 4 Completed by: <user who completed Post-Send Send Initial Notification Letter Holding Period>
> 5 Task Instructions: period of time in which task was on hold (e.g., 23/45 days) and <user input in the Proceed to final notification letter modal>

- [ ] The “Send Final Notification Letter” is created and active, task appears in the Currently Active 

> Tasks section is a blocking task for:
> *  Schedule hearing task for hearing docketed appeals.
> * Evidence submission window task for evidence docketed appeals.
> * Distribution task for direct docketed appeals.

>  Assigned to the COB organization

- [ ] Case appears on the Unassigned tab of the COB Queue

##### 2- The following occurs once the hold time expires for the “Post-Send Initial Notification Letter Holding Period” task

- [ ] “Post-Send Initial Notification Letter Holding Period” task is marked as Complete: Task moves from Currently Active Tasks section to Case Timeline and contains the following information
>1 Date of occurrence
> 2 Assigned to: <user or organization>
> 3 Assigned by: <user who completed action to assign task to COB>
> 4 Completed by: <user who completed Post-Send Send Initial Notification Letter Holding Period>
> 5 Task Instructions: period of time in which task was on hold (e.g., 45/45 days)

- [ ] “Send Final Notification Letter” task  is created and active
> 1 Task appears in the Currently Active Tasks section
> 2 Assigned to the COB organization
- [ ] Case appears on the Unassigned tab in the COB Queue

### Testing Plan 1

1. Go to Caseflow login and sign in as COB member.
2. Navigate to Queue and click on the dropdown in the right corner to switch views and select **Clerk of the Board team cases**
3. Moved to `On hold tab`, you can see all the appeals that are on hold

![c1](https://user-images.githubusercontent.com/110848569/221910207-4835e086-98f6-4096-89f9-7186b0a1b308.PNG)

4. Select one of them, and in the action dropdown select the option `Proceed to final notification Letter`

![c2](https://user-images.githubusercontent.com/110848569/221910783-c56b21da-baf7-4d50-b050-dc87c85741f8.PNG)

5. A modal windows has to show up, with a text area to put the description.

![c3](https://user-images.githubusercontent.com/110848569/221912315-812eafab-3377-41ed-a5d3-a4e4b5ec63f6.PNG)

6. Once you click in `Proceed to final letter` bottom, the appeal has to move to `Unassigned tab` on the table.

![c4](https://user-images.githubusercontent.com/110848569/221912407-24a4acd5-f179-4153-8ff2-78c2b6cf4153.PNG)

7. If you click in this appeal again and see the details page of this appeal, in the `Currently active task` area you can see the new task ` Send Final Notification Letter Task` for the appeal and a dropdown action. 

![c5](https://user-images.githubusercontent.com/110848569/221913151-a2e4591b-7042-474c-9832-d5bcf9961397.PNG)

8. In the `case Timeline` area you can see the previous task `PostSendInitialNotificationLetterHoldingTask` has to be mark as complete and show you Hold time period and a description.

![c6](https://user-images.githubusercontent.com/110848569/221913682-4b794c3b-db28-4404-9a05-7979eafc839d.PNG)

### Testing Plan 2 (hold time expires)

This part I testing using `.rspec file`, modified the current date 45 days forward and run the job to launch the process.
